docs/source/_notebooks/xgboost-titanic.rst

```diff
@@ -0,0 +1,2468 @@
+    from sklearn.utils import shuffle
+    from sklearn.model_selection import train_test_split
+    
+    feature_names = data[0][2:]
+    _all_xs = [dict(zip(feature_names, row[2:])) for row in data[1:]]
+    _all_ys = np.array([int(row[1]) for row in data[1:]])
+    
+    all_xs, all_ys = shuffle(_all_xs, _all_ys, random_state=0)
+    train_xs, valid_xs, train_ys, valid_ys = train_test_split(
+        all_xs, all_ys, test_size=0.25, random_state=0)
+    print('{} items total, {:.1%} true'.format(len(all_xs), np.mean(all_ys)))
+    pprint(valid_xs[:2])
+
+
+.. parsed-literal::
+
+    891 items total, 38.4% true
+    [{'Age': '19',
+      'Cabin': '',
+      'Embarked': 'S',
+      'Fare': '10.1708',
+      'Name': 'Dakic, Mr. Branko',
+      'Parch': '0',
+      'Pclass': '3',
+      'Sex': 'male',
+      'SibSp': '0',
+      'Ticket': '349228'},
+     {'Age': '19',
+      'Cabin': '',
+      'Embarked': 'Q',
+      'Fare': '7.8792',
+      'Name': 'Devaney, Miss. Margaret Delia',
+      'Parch': '0',
+      'Pclass': '3',
+      'Sex': 'female',
+      'SibSp': '0',
+      'Ticket': '330958'}]
+
+
+We do just minimal preprocessing: convert obviously contiuous ``Age``
+and ``Fare`` variables to floats, and ``SibSp``, ``Parch`` to integers.
+``Age`` can be missing, we default to 0 and will tell XGBoost to treat
+it as missing later.
+
+.. code:: python
+
+    for x in all_xs:
+        x['Age'] = float(x['Age'] or 0)
+        x['Fare'] = float(x['Fare'])
+        x['SibSp'] = int(x['SibSp'])
+        x['Parch'] = int(x['Parch'])
+
+2. Simple XGBoost classifier
+----------------------------
+
+Let's first build a very simple classifier with ``XGBClassifier`` and
+``sklearn.feature_extraction.DictVectorizer``, and check it's accuracy
+with cross-validation:
+
+.. code:: python
+
+    import warnings
+    warnings.filterwarnings('ignore', category=DeprecationWarning) 
+    from xgboost import XGBClassifier
+    from sklearn.feature_extraction import DictVectorizer
+    from sklearn.pipeline import make_pipeline
+    from sklearn.model_selection import cross_val_score
+    
+    clf = XGBClassifier(missing=0)
+    vec = DictVectorizer(sparse=False)  # https://github.com/dmlc/xgboost/issues/1238
+    pipeline = make_pipeline(vec, clf)
+    
+    def evaluate(_clf):
+        scores = cross_val_score(_clf, all_xs, all_ys, scoring='accuracy')
+        print('Accuracy: {:.3f} ± {:.3f}'.format(np.mean(scores), 2 * np.std(scores)))
+        _clf.fit(train_xs, train_ys)  # so that parts of the original pipeline are fitted
+         
+    evaluate(pipeline)
+
+
+.. parsed-literal::
+
+    Accuracy: 0.823 ± 0.008
+
+
+There are two tricky parts in above code. First is that we pass
+``missing=0`` to ``XGBClassifier``. This tells XGBoost to treat zeros as
+missing values, which is how most scikit-learn vectorizers work. It is
+important both for training and for correct feature visualization.
+
+Second tricky bit is that XGBClassifier has some
```

Comment:
Yes, hopefully :) I added xgboost version at the start of the tutorial with a link to their home page in d044c15